File: ciclos y arreglos.py
def calc_bmi(height_cm, weight_kg):
    """
    Calcula el índice de masa corporal (BMI).
    """
    height_m = height_cm / 100.0
    bmi = weight_kg / (height_m * height_m)
    return bmi


def main():
    # Pedir al usuario el número de personas
    num_users = int(input("Enter the number of users: "))


    # Listas para almacenar altura y peso de cada usuario
    heights = [0] * num_users
    weights = [0] * num_users
    bmis = [0] * num_users


    # Leer datos para cada usuario
    for i in range(num_users):
        print(f"User {i + 1}")


        # Pedir altura en centímetros
        heights[i] = float(input("Enter your height in cm: "))


        # Pedir peso en kilogramos
        weights[i] = float(input("Enter your weight in kg: "))


        # Calcular BMI para el usuario actual
        bmis[i] = calc_bmi(heights[i], weights[i])


    # Mostrar resultados para cada usuario
    print("\nBMI Results:")
    for i in range(num_users):
        print(f"User {i + 1}: BMI = {bmis[i]:.2f} kg/m^2")
       
        # Imprimir categoría del BMI
        if bmis[i] < 18.5:
            print(f"User {i + 1}: You are underweight")
        elif 18.5 <= bmis[i] < 24.9:
            print(f"User {i + 1}: Your weight is ideal")
        elif 25 <= bmis[i] < 29.9:
            print(f"User {i + 1}: You are overweight")
        elif 30 <= bmis[i] < 34.9:
            print(f"User {i + 1}: You have mild obesity")
        elif 35 <= bmis[i] < 39.9:
            print(f"User {i + 1}: You have moderate obesity")
        else:
            print(f"User {i + 1}: You are morbidly obese")


if __name__ == "__main__":
    main()

File: ciclos y arreglos.c
#include <stdio.h>


// Prototype of calcBmi function
float calcBmi(float height_cm, float weight_kg);


int main() {
    int numUsers;  // Number of users
    // Ask user for the number of users
    printf("Enter the number of users: ");
    scanf("%d", &numUsers);


    // Arrays to store height and weight for each user
    float heights[numUsers];
    float weights[numUsers];
    float bmis[numUsers];


    // Read data for each user
    for (int i = 0; i < numUsers; i++) {
        printf("User %d\n", i + 1);


        // Ask user for height in centimeters
        printf("Enter your height in cm: ");
        scanf("%f", &heights[i]);


        // Ask user for weight in kilograms
        printf("Enter your weight in kg: ");
        scanf("%f", &weights[i]);


        // Calculate BMI for the current user
        bmis[i] = calcBmi(heights[i], weights[i]);
    }


    // Display results for each user
    printf("\nBMI Results:\n");
    for (int i = 0; i < numUsers; i++) {
        printf("User %d: BMI = %.2f kg/m^2\n", i + 1, bmis[i]);
       
        // Print BMI category
        if (bmis[i] < 18.5) {
            printf("User %d: You are underweight\n", i + 1);
        } else if (bmis[i] >= 18.5 && bmis[i] < 24.9) {
            printf("User %d: Your weight is ideal\n", i + 1);
        } else if (bmis[i] >= 25 && bmis[i] < 29.9) {
            printf("User %d: You are overweight\n", i + 1);
        } else if (bmis[i] >= 30 && bmis[i] < 34.9) {
            printf("User %d: You have mild obesity\n", i + 1);
        } else if (bmis[i] >= 35 && bmis[i] < 39.9) {
            printf("User %d: You have moderate obesity\n", i + 1);
        } else if (bmis[i] >= 40) {
            printf("User %d: You are morbidly obese\n", i + 1);
        }
    }


    return 0; // Return 0 to indicate successful execution
}


// Implementation of calcBmi function
float calcBmi(float height_cm, float weight_kg) {
    float height_m = height_cm / 100.0;
    // Compute body mass index
    float bmi = weight_kg / (height_m * height_m);
    return bmi;
}
